proxy/tracing.go

```diff
@@ -67,40 +74,66 @@ func newProxyTracing(p *OpenTracingParams) *proxyTracing {
 	}
 
 	return &proxyTracing{
-		tracer:                   p.Tracer,
+		otTracer:                 p.OtTracer,
+		otelTracer:               p.OtelTracer,
 		initialOperationName:     p.InitialSpan,
 		disableFilterSpans:       p.DisableFilterSpans,
 		logFilterLifecycleEvents: p.LogFilterEvents,
 		logStreamEvents:          p.LogStreamEvents,
+		useOtel:                  p.UseNativeOtel,
 		excludeTags:              excludedTags,
 	}
 }
 
-func (t *proxyTracing) logEvent(span ot.Span, eventName, eventValue string) {
+func (t *proxyTracing) logEvent(span *tracing.SpanWrapper, eventName, eventValue string) {
 	if span == nil {
 		return
 	}
 
-	span.LogKV(eventName, ensureUTF8(eventValue))
+	if t.useOtel && span.Otel == nil {
+		return
+	}
+
+	if !t.useOtel && span.Ot == nil {
+		return
+	}
+
+	if t.useOtel {
+		span.Otel.AddEvent(
+			eventName,
+			otelTrace.WithAttributes(attribute.String(eventName, eventValue)),
+		)
+		return
+	}
+
+	span.Ot.LogKV(eventName, ensureUTF8(eventValue))
 }
 
-func (t *proxyTracing) setTag(span ot.Span, key string, value interface{}) *proxyTracing {
+func (t *proxyTracing) setTag(span *tracing.SpanWrapper, key string, value string) *proxyTracing {
 	if span == nil {
 		return t
 	}
 
+	if t.useOtel && span.Otel == nil {
+		return t
+	}
+
+	if !t.useOtel && span.Ot == nil {
+		return t
+	}
+
 	if !t.excludeTags[key] {
-		if s, ok := value.(string); ok {
-			span.SetTag(key, ensureUTF8(s))
+		if t.useOtel {
+			span.Otel.SetAttributes(attribute.String(key, ensureUTF8(value)))
 		} else {
-			span.SetTag(key, value)
+			span.Ot.SetTag(key, ensureUTF8(value))
```

Comment:
you can also implement `setTag()` in `(*tracing.SpanWrapper).setTag(k,v string)`